python/demo/gsoc2022/demo_sbc.py

```diff
@@ -0,0 +1,244 @@
+    gmsh.model.occ.addPlaneSurface([1], tag=1)
+
+    gmsh.model.occ.addCurveLoop([2], tag=2)
+    gmsh.model.occ.addCurveLoop([1], tag=3)
+    gmsh.model.occ.addPlaneSurface([2, 3], tag=2)
+
+    gmsh.model.occ.addCurveLoop([3], tag=4)
+    gmsh.model.occ.addCurveLoop([2], tag=5)
+    gmsh.model.occ.addPlaneSurface([4, 5], tag=3)
+
+    gmsh.model.occ.addCurveLoop([4], tag=6)
+    gmsh.model.occ.addCurveLoop([3], tag=7)
+    gmsh.model.occ.addPlaneSurface([6, 7], tag=4)
+
+    gmsh.model.occ.synchronize()
+
+    gmsh.model.addPhysicalGroup(2, [1, 2], tag=au_tag)
+    gmsh.model.addPhysicalGroup(2, [3, 4], tag=bkg_tag)
+
+    gmsh.model.addPhysicalGroup(1, [4], tag=boundary_tag)
+
+    gmsh.model.mesh.setSize([(0, 1)], size=in_wire_size)
+    gmsh.model.mesh.setSize([(0, 2)], size=on_wire_size)
+    gmsh.model.mesh.setSize([(0, 3)], size=bkg_size)
+    gmsh.model.mesh.setSize([(0, 4)], size=boundary_size)
+
+    gmsh.model.mesh.generate(2)
+
+mesh, cell_tags, facet_tags = gmsh_model_to_mesh(gmsh.model, cell_data=True, facet_data=True, gdim=2)
+
+MPI.COMM_WORLD.barrier()
+
+# definition of finite element for the electric field
+curl_el = ufl.FiniteElement("N1curl", mesh.ufl_cell(), 3)
+
+# definition of finite element for the r function (see next)
+lagr_el = ufl.FiniteElement("CG", mesh.ufl_cell(), 2)
+
+# function space for the electric field
+V = fem.FunctionSpace(mesh, curl_el)
+
+# wavevector of the background field
+k0 = 2 * np.pi / wl0
+
+# angle of incidence of the background field with respect to the horizontal axis
+deg = np.pi / 180
+theta = 45 * deg
+
+# plane wave function
+f = background_electric_field(theta, n_bkg, k0)
+Eb = fem.Function(V, dtype=np.complex128)
+Eb.interpolate(f.eval)
+
+# function r = radial distance from the (0, 0) point
+lagr_space = fem.FunctionSpace(mesh, lagr_el)
+r = fem.Function(lagr_space, dtype=np.complex128)
+r.interpolate(radial_distance)
+
+# definition of Trial and Test functions
+Es = ufl.TrialFunction(V)
+Vs = ufl.TestFunction(V)
+
+# definition of 3d fields for cross and curl operations
+Es_3d = from_2d_to_3d(Es)
+Vs_3d = from_2d_to_3d(Vs)
+
+# Measures for subdomains
+dAu = ufl.Measure("dx", mesh, subdomain_data=cell_tags, subdomain_id=au_tag)
+dBkg = ufl.Measure("dx", mesh, subdomain_data=cell_tags, subdomain_id=bkg_tag)
+dsbc = ufl.Measure("ds", mesh, subdomain_data=facet_tags, subdomain_id=boundary_tag)
+dDom = dAu + dBkg
+
+# normal to the boundary
+n = FacetNormal(mesh)
+n_3d = from_2d_to_3d(n)
+
+# definition of relative permittivity for Au @400nm
+reps_au = -1.0782
+ieps_au = 5.8089
+eps_au = reps_au + ieps_au * 1j
+
+# definition of the relative permittivity over the whole domain
+D = fem.FunctionSpace(mesh, ("DG", 0))
+eps = fem.Function(D)
+au_cells = cell_tags.indices[cell_tags.values == au_tag]
+bkg_cells = cell_tags.indices[cell_tags.values == bkg_tag]
+eps.x.array[au_cells] = np.full(len(au_cells), reps_au + ieps_au * 1j)
+eps.x.array[bkg_cells] = np.full(len(bkg_cells), eps_bkg)
```

Comment:
```suggestion
eps.x.array[au_cells] = np.full_like(au_cells, reps_au + ieps_au * 1j)
eps.x.array[bkg_cells] = np.full_like(bkg_cells, eps_bkg)
```

Reply:
I needed to add `dtype=np.complex128` to make it run.

Reply:
i.e. 

```
eps.x.array[au_cells] = np.full_like(au_cells, reps_au + ieps_au * 1j, dtype=np.complex128)
eps.x.array[bkg_cells] = np.full_like(bkg_cells, eps_bkg, dtype=np.complex128)
```